src/guid.rs

```diff
@@ -0,0 +1,123 @@
+use std::{fmt, str, ops};
+
+// TODO: We could store a 13th b'\0' byte for a C-compatible null terminator. Is
+// this worth doing?
```

Comment:
I don't think so...that would be useful if we're exposing trees over the FFI, right? Or just to make conversions to and from strings easier?

Reply:
It would be to allow exposing the memory from one of these as a null terminated C string without requiring a heap allocation. It's probably not needed?.